src/Http/Client.php

```diff
@@ -168,23 +169,45 @@ public function __construct($config = [])
             $this->_cookies = $this->_config['cookieJar'];
             $this->setConfig('cookieJar', null);
         } else {
-            $this->_cookies = new CookieCollection();
+            $this->_cookies = new ClientCookieCollection();
         }
     }
 
     /**
      * Get the cookies stored in the Client.
      *
-     * Returns an array of cookie data arrays.
-     *
-     * @return \Cake\Http\Client\CookieCollection
+     * @return \Cake\Http\Cookie\CookieCollection
+     * @deprecated 3.5.0 Use getCookies() instead.
      */
     public function cookies()
     {
         return $this->_cookies;
     }
 
     /**
+     * Get the cookies stored in the Client.
+     *
+     * @return \Cake\Http\Cookie\CookieCollection
+     */
+    public function getCookies()
+    {
+        return $this->_cookies;
+    }
+
+    /**
+     * Sets the cookie collection.
+     *
+     * @param \Cake\Http\Cookie\CookieCollection $cookies Cookie collection to be set.
+     * @return $this
+     */
+    public function setCookies(CookieCollection $cookies)
```

Comment:
Keeping the class small and focuses on features they need is the only reason I can think of.